### ISSUE TITLE
FAWE schematic restoration incompatibility?

### ISSUE THREAD

**Issue by acidbyt3** [NLthijs48/AreaShop#188]
#### Current behavior
Restoring a region from a schematic doesn't work well with Fast Async WorldEdit (FAWE). The paste is erratic, and there are errors thrown onto the console: http://pastebin.com/2pPem3DX

#### New behavior
FAWE compatibility would be greatly appreciated :)

#### Steps to reproduce (for bugs)
1. Run AreaShop alongside FAWE with schematic restoration enabled
2. Create a region/add as an area shop
3. Buy and sell it back, manually force the /as schematicevent, or any other way to trigger the schematic restore event
4. The paste result is incorrect, terrain is not reverted properly.

#### Your environment
* Server platform: git-Spigot-078e2f8-7f9fbe5 (Spigot)
* Server version: 1.10.2
* AreaShop version: 2.2.2 (as per the config.yml, couldn't find the ingame command)
* FAWE version: 3.5.1-2f76cb5
* WorldEdit version: 6.1.4-SNAPSHOT;3595-c957ffb
* WorldGuard version: 6.1.2-SNAPSHOT;1664-5bd2125
* Vault version: 1.5.6-b49

**Comment by NLthijs48** [comment]
I'm just using some WorldEdit functions to past the schematic, so it looks like FAWE broke one of those methods. I could add extra code to paste the schematic through FAWE, however if that can be avoided then that would be better. Did you also report this to FAWE? Because this might actually be something that they can fix, considering it is the `DiskOptimizedClipboard` class of FAWE that is reading a file in a wrong way. You might also try to disable disk storage for clipboards, and just keep them in memory if you have that feature of FAWE enabled, that might fix the problem.

**Comment by acidbyt3** [comment]
Apologies, I ended up forgetting to update this.
It seems there is an issue with that FAWE version when pasting schematics in general, I noticed this the other day when multiple users were having trouble pasting schematics. Originally I had assumed it was an incompatibility because it was the first time I was trying your plugin and happened to update FAWE at the same time, and never had pasting issues with FAWE before. I downgraded FAWE to the previous version and schematics started pasting as normal again.

I haven't tested AreaShop since (haven't had much time) but I think it's safe to assume that the issue was related and should now be fixed. I'll report back if the problem persists, when I get a chance to test it again.

I apologise for the false alarm! Thanks for your help :)
